tests/test_views.py

```diff
@@ -0,0 +1,455 @@
+
+    def test_logged_in_user_access(self) -> None:
+        for url in (
+                DRIVER_LIST_URL,
+                DRIVER_CREATE_URL,
+                get_driver_detail_url(self.driver),
+                get_driver_update_url(self.driver),
+                get_driver_delete_url(self.driver)
+        ):
+            http_response = self.client.get(url)
+            self.assertEqual(
+                http_response.status_code,
+                200
+            )
+
+    def test_list(self) -> None:
+        http_response = self.client.get(DRIVER_LIST_URL)
+        self.assertEqual(
+            list(http_response.context["driver_list"]),
+            list(self.all_drivers)
+        )
+
+    def test_list_search(self) -> None:
+        for search_letter in ("1", "2", "t"):
+            http_response = self.client.get(
+                DRIVER_LIST_URL,
+                {"username": f"{search_letter}"}
+            )
+            self.assertEqual(
+                list(http_response.context["driver_list"]),
+                list(self.all_drivers.filter(
+                    username__icontains=f"{search_letter}")
+                )
+            )
+
+    def test_detail(self) -> None:
+        self.assertEqual(
+            self.client.get(
+                get_driver_detail_url(self.user)).context["driver"],
+            self.user
+        )
+        self.assertEqual(
+            self.client.get(
+                get_driver_detail_url(self.driver)).context["driver"],
+            self.driver
+        )
+
+    def test_create(self) -> None:
+        http_response = self.client.post(
+            DRIVER_CREATE_URL,
+            data=self.driver_data
+        )
+        new_driver = get_user_model().objects.get(
+            username=self.driver_data["username"]
+        )
+        self.assertEqual(
+            http_response.status_code,
+            302
+        )
+        self.assertEqual(
+            new_driver.license_number,
+            self.driver_data["license_number"]
+        )
+        self.assertTrue(
+            new_driver.check_password(self.driver_data["password1"])
+        )
+
+    def test_update(self) -> None:
+        http_response = self.client.post(
+            get_driver_update_url(self.driver),
+            data={
+                "license_number": self.driver_data["license_number"]
+            }
+        )
+        new_driver = get_user_model().objects.get(
+            username=self.driver.username
+        )
+        self.assertEqual(
+            http_response.status_code,
+            302
+        )
+        self.assertEqual(
+            new_driver.license_number,
+            self.driver_data["license_number"]
+        )
+
+    def test_delete(self) -> None:
+        http_response = self.client.post(
+            get_driver_delete_url(self.driver)
+        )
+        self.assertEqual(
+            http_response.status_code,
+            302
+        )
+        self.assertEqual(
```

Comment:
In the `test_delete` method of `DriverPrivateTest`, consider verifying that the driver objects are actually deleted from the database by checking that they cannot be retrieved anymore, rather than just checking the length of the queryset.